src/utils/ReadOnlyError.ts

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { InternalException } from './InternalException';
+
+export class ReadOnlyError extends InternalException implements TypeError {
```

Comment:
This should be `extends TypeError` so that if people do `err instanceof TypeError` it'll return `true`. It's a built-in JavaScript/Node.js class to be used "when attempting to modify a value that cannot be changed" per [these docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError)

Since a class can only extend one other class you might have to copy the `isAzureFunctionsInternalException` property, but I think that's fine if it means we use standard Node.js classes.

Reply:
Yeah I was trying to `extend` first but because I can only extend one class I did the `implements` instead. Changed to your suggestion